### PR TITLE
feat: support custom menu render functions in Select

### PR DESCRIPTION
**Pull request by connorjsmith**
# Description
This change allows for custom menu render functions in the `Select` component. This mirrors the existing custom `trigger` render function allows for a user to override the default trigger.

## Motivation & context
Allow for more flexibility in the menu rendered by a `Select`, including wrapping the default menu in a background provider as described in #2250  and #2266

## Issue type checklist

- [ ] **Chore**: A change that does not impact distributed packages.
- [ ] **Bug fix**: A change that fixes an issue, link to the issue above.
- [x] **New feature**: A change that adds functionality.

**Is this a breaking change?**
- [ ] This change causes current functionality to break.

## Process & policy checklist

- [x] I have added tests for my changes.
- [x] I have tested my changes.
- [x] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast-dna/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/en/contributing/standards) for this project.

### CODE REVIEW

**Comment by bheston** [comment]
Looks good to me. Feel like fixing the msft style control, or in another PR?

**Comment by connorjsmith** [comment]
Forgot the msft component isn't a straight manageJss call. Will update that in a sec

File: packages/fast-components-react-msft/src/select/select.tsx [hunk 1]
```diff
@@ -26,6 +27,7 @@ class Select extends Foundation<SelectHandledProps, SelectUnhandledProps, {}> {
                 {...this.unhandledProps()}
                 managedClasses={this.props.managedClasses}
                 disabled={this.props.disabled}
+                menu={this.props.menu}
```
Comment (bheston) [inline packages/fast-components-react-msft/src/select/select.tsx]: This doesn't quite resolve 2250, but it lets you fix it in your code. We should address it as the default stying though. Can you wrap the default menu in a Background and add neutralLayerFloating for its color?

**Comment by janechu** [comment]
@connorjsmith looks like you need to run `lerna run prettier` in the  to do our auto-formatting, for some reason our npm hooks which should automatically add that on commit aren't quite working

**Pull request merged**